src/lxml/apihelpers.pxi

```diff
@@ -564,11 +564,14 @@ cdef int _setAttributeValue(_Element element, key, value) except -1:
     if not element._doc._parser._for_html:
         _attributeValidOrRaise(tag)
     c_tag = _xcstr(tag)
-    if isinstance(value, QName):
-        value = _resolveQNameText(element, value)
+    if value is None:
```

Comment:
Shouldn't it reject `None` for XML trees and only allow it for HTML trees? The latter is the only use case I can see. XML always requires at least an empty string as value.